lib/trino-orc/src/main/java/io/trino/orc/OutputStreamOrcDataSink.java

```diff
@@ -30,10 +33,28 @@
     private static final int INSTANCE_SIZE = toIntExact(ClassLayout.parseClass(OutputStreamOrcDataSink.class).instanceSize());
 
     private final OutputStreamSliceOutput output;
+    private final AggregatedMemoryContext memoryContext;
 
-    public OutputStreamOrcDataSink(OutputStream outputStream)
+    private OutputStreamOrcDataSink(OutputStream outputStream, AggregatedMemoryContext memoryContext)
     {
         this.output = new OutputStreamSliceOutput(requireNonNull(outputStream, "outputStream is null"));
+        this.memoryContext = requireNonNull(memoryContext, "memoryContext is null");
+    }
+
+    public static OutputStreamOrcDataSink create(TrinoOutputFile outputFile, AggregatedMemoryContext memoryContext)
+            throws IOException
+    {
```

Comment:
change the method from `create(TrinoOutputFile outputFile, AggregatedMemoryContext memoryContext)` to `create(TrinoOutputFile outputFile)` and create memory context internally. You don't need external memory context since you account for mem in `getRetainedSizeInBytes`